osmpy/core.py

```diff
@@ -149,12 +155,24 @@ def get(query, boundary, threshold_value=1000000):
         if query in _get_queries_names():
             query_obj = getattr(sys.modules['osmpy.queries'], query)()
         else:
-            query_obj = qs.QueryType()
+            query_obj = QueryType()
             query_obj.query = query
     elif isinstance(query, type):
         query_obj = query()
 
-    boundaries = katana(boundary, threshold_func, threshold_value)
+    boundaries = katana(boundary, threshold_func, threshold_value, urllen_threshold=urllen_threshold)
+    
+    # Looking for the boundaries which generate too long URLs resulting in Error 414.
+    # If found the boundaries will be processed again by `katana()`
+    while True:
```

Comment:
shouldn't katana itself deal with that? 

If `urllen > threshold` then it goes into it again.

Reply:
I started by adding `urrlen_threshold` to `katana()` so that the function could decide if it needed to split the boundaries. However, later I encountered rare cases where even 250 levels of splitting were not enough to generate a short enough URL, which resulted in Error 414. To prevent long URLs from appearing again, I added this part to ensure that the boundaries will always be split enough.